x-pack/plugins/apm/server/routes/agent_explorer/get_agents_filters.ts

```diff
@@ -0,0 +1,96 @@
+import { AGENT_NAME, AGENT_VERSION, SERVICE_LANGUAGE_NAME, SERVICE_NAME, SERVICE_NODE_NAME } from "@kbn/apm-plugin/common/elasticsearch_fieldnames";
+import { environmentQuery } from "@kbn/apm-plugin/common/utils/environment_query";
+import { ProcessorEvent } from "@kbn/observability-plugin/common";
+import { rangeQuery } from "@kbn/observability-plugin/server";
+import { RandomSampler } from "../../lib/helpers/get_random_sampler";
+import { Setup } from "../../lib/helpers/setup_request";
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+export async function getAgentsFilters({
+  environment,
+  setup,
+  start,
+  end,
+  randomSampler,
+}: {
+  environment: string;
+  setup: Setup;
+  start: number;
+  end: number;
+  randomSampler: RandomSampler;
+}) {
+const { apmEventClient } = setup;
+
+const response = await apmEventClient.search(
+  'get_agents_services',
+  {
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      track_total_hits: false,
+      size: 0,
+      query: {
+        bool: {
+          filter: [
```

Comment:
Are these filters needed? It might be good for the user to be able to filter by all available  services and then find out that there might be missing information for agents. You might also consider splitting the filter requests